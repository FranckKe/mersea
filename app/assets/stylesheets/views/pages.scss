main {
  height: 100%;
}

.map {
  width: 100%;
  height: 100%;

  .leaflet-bottom.leaflet-right .leaflet-control-layers {
    max-height: calc(100vh - 55px - 74px - 10px - 32px - 5px); // Menu - Zoom Buttons - Bottom Margin - Footer - 5px
    overflow-y: scroll;
  }
}

.leaflet-marker-icon {
  max-width: 64px;
  max-height: 64px;
}

.article {
  width: 100%;
  max-width: 750px;
  margin: auto;
  padding-top: 25px;
  padding-right: 25px;
  padding-left: 25px;

  h1 {
    text-align: center;
  }
}
