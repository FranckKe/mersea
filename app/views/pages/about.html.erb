<div class="page-container">
    <h1>Description du projet</h1>
    <h2>Les impacts de la pollution plastique</h2>
    <p>8 millions de tonnes de plastiques sont déversés dans les océans chaque année. Dommages sur les habitants, perturbateurs
        endocriniens, blessures, décès d’animaux marins par ingestion ou étranglement, ces pollutions massives entraînent
        des pertes économiques majeures (baisse du tourisme, impacts sur les pêches…).</p>

    <p>80% des déchets retrouvés sur le littoral proviennent du continent par les rivières, les eaux pluviales. Le reste provient
        du trafic maritime et des activités en mer. On estime à 10 000 le nombre de conteneurs perdus en mer par an au niveau
        mondial.
    </p>

    <p>Pertes économiques (baisse tourisme), dommages sur les habitats marins, blessures ou décès d’animaux marins par ingestion
        ou étranglement. Des problèmes pour la navigation, déplacement d'espèces marines qui s'accroche aux déchets (<a href="https://marinedebris.noaa.gov/discover-issue/impactsn">www.marinedebris.noaa.gov</a>).</p>

    <h2>Le rôle des traceurs</h2>

    <p>Parmi les pollutions plastiques, certains déchets racontent une histoire et permettent de suivre leur propagation au
        gré des courants et des tempêtes. Ces traceurs permettent de qualifier et de quantifier une pollution provenant soit
        de conteneurs perdus en mer soit de déversements accidentels en milieu naturel. Avec plus de 150 navires qui passent
        tous les jours dans le rail d’Ouessant, la zone Atlantique est un observatoire unique de ces traceurs.
    </p>

    <h2>Technologie</h2>
    <p>Le code est publié sous licence Open Source MIT sur GitHub : Les contributeurs au code : Pierre Charpentier / Marc Douchement
        / Franck Kerbiriou</p>
</div>