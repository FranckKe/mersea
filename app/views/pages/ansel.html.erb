<div class="page-container">
    <h1>ANSEL & co</h1>
    <p>L’association ANSEL sensibilise aux échouages de déchets sur le littoral depuis 17 ans. Après avoir lancé le Réseau d’Observation
        d'Échouages de Déchets en Atlantique en 2014, le besoin d’une cartographie devient évident pour suivre les nombreuses
        observations collectées. Les premières réflexions avec le fonds Explore de Roland Jourdain sur ce projet de carto
        se concrétise lors du Ocean Hackathon de Brest en octobre 2016. A cette occasion Pierre Charpentier, Franck Kerbiriou,
        Marc Douchement et Caroline Geisert développent en 48h la première version de la cartographie.
    </p>

    <h2>Contacts</h2>
    <p>Ansel asso 29900 Concarneau</p>
    <p>Réseau d’observation d’échouage de déchets en atlantique, Facebook.</p>
</div>