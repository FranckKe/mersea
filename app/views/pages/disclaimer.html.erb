<div class="page-container">
    <h2>Mentions légales</h2>
    <p>L’association ANSEL est producteur de la base de donnée issue des observations. Les données que vous communiquez seront
        exploitées exclusivement dans le cadre des objectif de ce projet. Les photos communiquées sont sous la responsabilité
        du producteur et sont temporairement stockées pour la validation des relevés. En communiquant la photographie pour
        la réalisation de ce projet, le producteur cède les droits d’utilisation au profit de la plateforme.</p>
    <p>Conformément à la loi "informatique et libertés" du 6 janvier 1978 modifiée, vous bénéficiez d’un droit d’accès et de
        rectification aux informations qui vous concernent. Si vous souhaitez exercer ce droit et obtenir communication des
        informations vous concernant, veuillez vous adresser à l’association ANSEL.
    </p>
</div>