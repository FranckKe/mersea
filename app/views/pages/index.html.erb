<main>
  <div id="map" class="map"></div>
</main>
