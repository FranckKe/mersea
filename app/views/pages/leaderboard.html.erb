<h2 class="text-center">Palmarès</h2>
<section class="flex user-reports small-12 medium-10 medium-centered">
  <table class="palmares__table">
    <thead>
      <th>Nom</th>
      <th>Nombre de témoignages</th>
      <th>Dernière activité</th>
    </thead>
    <tbody>
    <% @leaderboard.each do |user| %>
      <tr>
        <td><%= user.name %></td>
        <td><%= user.reports.count %></td>
        <td><%= user.reports.last.reported_at.strftime("%d-%m-%Y") %></td>
      </tr>
    <% end %>
    </tbody>
  </table>
</section>
