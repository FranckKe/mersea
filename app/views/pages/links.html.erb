<div class="page-container">
    <h1>Liens utiles</h1>
    <ul>
        <li>Mammifères et oiseaux échoués <a href="http://crmm.univ-lr.fr/index.php/fr/reseau-national-echouages" target="_blank">Réseau Pélagis</a></li>
        <li>Méduses échouées <a href="http://www.mer-littoral.org/formulaire-meduses.php" target="_blank">Réseau “Mer et Littoral”</a></li>
        <li>Bloom planctonique <a href="http://www.phenomer.org/" target="_blank">Phenomer</a></li>
        <li>Hydrocarbures et explosifs : appeler les pompiers</li>
    </ul>
</div>
