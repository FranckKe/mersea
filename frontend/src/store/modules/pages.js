import axios from 'axios'
import i18n from '../../i18n'
import slugify from 'slugify'

const api = axios.create({
  baseURL: process.env.VUE_APP_API_URL,
  headers: { Accept: 'application/json', 'Content-Type': 'application/json' }
})

const state = {
  pages: [],
  loading: true,
  success: false,
  errors: []
}
const getters = {
  getLoading: state => {
    return state.loading
  },
  getError: state => {
    return state.errors
  },
  getSuccess: state => {
    return state.success
  },
  getPageContent: state => {
    return pageInfo => {
      return (
        state.pages.filter(page => page.slug === pageInfo.pageName)[0] || {}
      ).content
    }
  },
  getCategories: state => {
    return [
      ...new Set(
        state.pages
          .map(page => page.category)
          .filter(page => page.category !== 'other')
      )
    ]
  },
  getAllPagesByCategory: state => {
    let categories = [
      ...new Set(
        state.pages
          .map(page => page.category)
          .filter(page => page.category !== 'other')
      )
    ]
    return categories.map(function(category) {
      return {
        category: slugify(category),
        pagesName: state.pages
          .filter(
            page => page.category === category && page.language === i18n.locale
          )
          .map(page => {
            return { raw: page.name, slug: page.slug }
          })
      }
    })
  },
  getPagesByCategory(state) {
    return category => {
      return state.pages
        .filter(
          page => page.category === category && page.language === i18n.locale
        )
        .map(page => {
          return { raw: page.name, slug: page.slug }
        })
    }
  }
}
const mutations = {
  setPages: (state, { pages }) => {
    state.pages = pages
  },
  setSuccess: state => {
    state.success = true
    state.loading = false
    state.errors = []
  },
  setError: (state, { errors }) => {
    state.errors = errors
    state.loading = false
  }
}
const actions = {
  async loadPages({ commit }) {
    try {
      const pagesData = await api.get(`/pages`)
      const pages = pagesData.data
      commit('setPages', { pages })
      commit('setSuccess')
    } catch (e) {
      let errors = [e.message]
      commit('setError', { errors })
    }
  }
}

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
}
