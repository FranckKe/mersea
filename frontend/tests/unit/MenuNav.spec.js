import { shallowMount, createLocalVue } from '@vue/test-utils'
import Menu from '@/components/MenuNav.vue'
import Vuex from 'vuex'
import pages from '../../src/store/modules/pages'
import Buefy from 'buefy'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(Buefy)
localVue.use(Vuex)
localVue.use(VueRouter)

const router = new VueRouter()

describe('Page.vue', () => {
  let store

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        pages: pages
      }
    })
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(Menu, { store, localVue, router })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
