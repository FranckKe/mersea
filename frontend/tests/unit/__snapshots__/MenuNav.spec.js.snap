// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page.vue matches snapshot 1`] = `
<nav role="navigation" aria-label="main navigation" class="navbar is-fixed-top">
  <bloading-stub active="true" animation="fade" cancancel="true" oncancel="function (){}"></bloading-stub>
  <div class="navbar-brand">
    <div class="navbar-start">
      <a role="button" aria-label="menu" data-target="flexible-menu" aria-expanded="false" class="navbar-burger"><span aria-hidden="true"></span> <span aria-hidden="true"></span> <span aria-hidden="true"></span></a>
    </div>
  </div>
  <div id="flexible-menu" class="navbar-menu">
    <router-link-stub to="/" tag="a" event="click" class="navbar-item">Home</router-link-stub>
    <router-link-stub to="/tracers" tag="a" event="click" class="navbar-item">Tracers</router-link-stub>
    <router-link-stub to="/leaderboard" tag="a" event="click" class="navbar-item">Leaderboard</router-link-stub>
  </div>
</nav>
`;
